### DISCUSSION TITLE
Type safe conversion from XML to Typescript object

### DISCUSSION THREAD

**Discussion by belostoky** [oozcitak/xmlbuilder2#139] [Q&A]
I'm trying to use xmlbuilder2 to convert XML string to Typescript object.
I know how to convert it to json, and then cast to my object like such:
`JSON.parse(convert(xmlStr2, { format: 'json' })) as MyType;`
Hoping I can convert it to `MyType` directly, without going through `JSON.parse`.
Is it possible?
Thanks

**Comment by oozcitak** [comment]
Not tested but should work:
```js
convert(xmlStr2, { format: 'object' }) as MyType;
```
See: https://oozcitak.github.io/xmlbuilder2/serialization.html#js-object and https://oozcitak.github.io/xmlbuilder2/serialization.html#serialization-settings
